plugins/modules/clustering/consul/consul.py

```diff
@@ -589,15 +573,27 @@ def main():
             token=dict(no_log=True)
         ),
         supports_check_mode=False,
+        mutually_exclusive=[
+            ('script', 'ttl', 'tcp', 'http'),
+        ],
+        required_by={
+            'script': 'interval',
+            'http': 'interval',
+            'tcp': 'interval',
```

Comment:
Same here, these dependencies were only required if `state=present`.

Reply:
None of these parameters are used when `state=absent`.